__tests__/app.test.js

```diff
@@ -284,7 +284,131 @@ describe("get comments by articles id", () => {
+    const body = {
+      username: "lurker",
+      body: "robots are robots because robot makes robot",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(body)
+      .expect(201)
+      .then((response) => {
+        const { comment } = response.body;
+        expect(comment).toHaveProperty("article_id", 1);
+        expect(comment).toHaveProperty("author", "lurker");
+        expect(comment).toHaveProperty(
+          "body",
+          "robots are robots because robot makes robot"
+        );
+        expect(comment).toHaveProperty("comment_id", 19);
+        expect(comment).toHaveProperty("votes", 0);
+      });
+  });
+  test("should respond with object with correct properties", () => {
+    const body = {
+      username: "lurker",
+      body: "robots are robots because robot makes robot",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(body)
+      .expect(201)
+      .then((response) => {
+        return request(app)
+          .get("/api/articles/1/comments")
+          .expect(200)
+          .then((getResponse) => {
+            const { comments } = getResponse.body;
+            expect(Array.isArray(comments)).toBe(true);
+            expect(comments.length).toBe(12);
+          });
+      });
+  });
+
+  test("should respond with 404 for valid but non existent article request", () => {
+    const body = {
+      username: "lurker",
+      body: "robots are robots because robot makes robot",
+    };
+    return request(app)
+      .post("/api/articles/1000/comments")
+      .send(body)
+      .expect(404)
+      .then((response) => {
+        const { msg } = response.body;
+        expect(msg).toBe("article not found");
+      });
+  });
+
+  test("should respond with 400 error for invalid article request", () => {
+    const body = {
+      username: "lurker",
+      body: "robots are robots because robot makes robot",
+    };
+    return request(app)
+      .post("/api/articles/robot/comments")
+      .send(body)
+      .expect(400)
+      .then((response) => {
+        const { msg } = response.body;
+        expect(msg).toBe("bad request");
+      });
+  });
+
+  test("should respond with 400 error for invalid body", () => {
+    const body = {
+      username: "lurker",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(body)
+      .expect(400)
+      .then((response) => {
+        const { msg } = response.body;
+        expect(msg).toBe("bad request");
+      });
+  });
+
+  test("should respond with 400 error for invalid username", () => {
+    const body = {
+      username: "robot",
+      body: "robots are robots because robot makes robot",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(body)
+      .expect(400)
+      .then((response) => {
+        const { msg } = response.body;
+        expect(msg).toBe("bad request");
```

Comment:
It would be great to see you add another test for 
Status 201, ignores unnecessary properties, for instance if you sent 
```
const body = {
      username: "robot",
      body: "robots are robots because robot makes robot",
      nickname: 'hannah'
    };
    
```
    It would just ignore the nickname property.